src/widget.cpp

```diff
@@ -511,6 +547,16 @@ std::string widget::color_text_function_string( const avatar &ava )
         case widget_var::wind_text:
             desc = display::wind_text_color( ava );
             break;
+        case widget_var::compass_text:
+            // Compass color is specific to individual threats.
+            // Skip colorization.
+            desc = display::compass_text_color( _direction, _width );
```

Comment:
I think I see what you did here - the `compass_text_color` returns a `std::string` that already embeds colorized strings for each threat, and the second `nc_color` part is just a (dummy?) `c_white` that ends up being ignored.

There is a little bit of precedent in `panels.h` in the `display` namespace for functions that return *only* a `std::string` - sometimes even colorized, like `display::get_moon_graphic`. So I think you could change `compass_legend_text_color` to return an already-colorized string (instead of a string/color pair), and change its name to something like `colorized_compass_legend_text`, to avoid this little hack of ignoring the second part of the pair.

I'm also not crazy about doing a `return` from the middle of this switch/case statement; I would propose using a flag, like:

```cpp
bool do_colorize = true;
switch( _var ) {
    // ...
    case widget_var::compass_text:
        desc.first = display::colorized_compass_legend_text( _direction, _width );
        do_colorize = false;
        break;
    // case ...
}
if( do_colorize ) {
    ret += colorize( desc.first, desc.second );
} else {
    ret += desc.first; // assume already colorized
}
return ret;
```

I expect this to come up again (in fact I was thinking about it for the vehicle and bodypart widgets too), and have a feeling this would give us a cleaner logical flow for this exceptional case.

Reply:
No problem, I didn't realize I was breaking the pattern, haha.

I changed the return types for the compass display functions and removed the early returns. I'll try to stick to your format here going forward (sorry!).